### PR TITLE
SVM sigmoid kernel fix (issue #13621)

### LINKED ISSUE #13621: Bug in SVM sigmoid kernel
Normally sigmoid kernel used in SVM is:
K(x, y) = tanh(gamma * <x, y> + coef0)

But in this  code different formula is implemented:
K(x, y) = -tanh(gamma * <x, y> + coef0)
https://github.com/opencv/opencv/blob/7d4bb9428b31ad144cfdc8cafe26b5c37258d489/modules/ml/src/svm.cpp#L198-L214

I'm not sure if it is intended, looks like a bug for me.

### PR DESCRIPTION
**Pull request by lochsh**
resolves #13621

### This pullrequest changes
_tl;dr sigmoid kernel is now as it should be, removed a restriction on coef0 that I understand to be incorrect_

Normally the sigmoid kernel used in SVM is:
K(x, y) = tanh(gamma * <x, y> + coef0)

But OpenCV had implemented the negative of this:
K(x, y) = -tanh(gamma * <x, y> + coef0)

[The OpenCV documentation](https://docs.opencv.org/3.4/d1/d2d/classcv_1_1ml_1_1SVM.html#aad7f1aaccced3c33bb256640910a0e56) says that it should be the former.

I've fixed this and added a test to give some coverage.  I chose a simple dataset that seemed well suited to the sigmoid kernel, and saw that someone else had used successfully:
image

When trying to replicate the gamma and coef0 values used in this example, I found that coef0 was restricted to be 0 or positive, but that isn't a necessary or sensible restriction given my understanding of the maths. Python's sci-kit learn doesn't enforce it.

I removed this restriction also.

This is my first time contributing to OpenCV and I'd appreciate feedback -- I wasn't sure if I was taking the right approach with the test, and tried to follow from what already existed.

```
buildworker:Win64 OpenCL=windows-2
```

### CODE REVIEW

File: modules/ml/test/test_svmtrainauto.cpp [hunk 1]
```diff
@@ -88,6 +88,51 @@ void CV_SVMTrainAutoTest::run( int /*start_from*/ )
...
+    cv::Ptr<SVM> svm = SVM::create();
+    svm->setKernel(SVM::SIGMOID);
+
+    svm->setGamma(10.0);
+    svm->setCoef0(-10.0);
+    svm->trainAuto( data, 10 );  // 2-fold cross validation.
+
+    float test_data0[2] = {radius, radius};
+    cv::Mat test_point0 = cv::Mat( 1, 2, CV_32FC1, test_data0 );
+    CV_Assert(0 == svm->predict( test_point0 ));
```
[the remark this answers is not in the document]
Comment (lochsh) [reply modules/ml/test/test_svmtrainauto.cpp]: Right on, I've sorted this now.
